PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = "product.category"
-    _inherit = 'res.partner'
-    _inherit = 'sale.order'
-    _inherit = 'sale.order.line'

Total Changes: 13

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/delivery/models/delivery_carrier.py
+++ b/addons/delivery/models/delivery_carrier.py
@@ -10,7 +10,6 @@ from odoo.tools.safe_eval import safe_eval
 
 
 class DeliveryCarrier(models.Model):
-    _name = 'delivery.carrier'
     _description = "Shipping Methods"
     _order = 'sequence, id'
 

--- a/addons/delivery/models/delivery_price_rule.py
+++ b/addons/delivery/models/delivery_price_rule.py
@@ -13,8 +13,7 @@ VARIABLE_SELECTION = [
 ]
 
 
-class PriceRule(models.Model):
-    _name = "delivery.price.rule"
+class DeliveryPriceRule(models.Model):
     _description = "Delivery Price Rules"
     _order = 'sequence, list_price, id'
 

--- a/addons/delivery/models/delivery_zip_prefix.py
+++ b/addons/delivery/models/delivery_zip_prefix.py
@@ -5,7 +5,6 @@ from odoo import api, fields, models
 
 class DeliveryZipPrefix(models.Model):
     """ Zip prefix that a delivery.carrier will deliver to. """
-    _name = 'delivery.zip.prefix'
     _description = 'Delivery Zip Prefix'
     _order = 'name, id'
 

--- a/addons/delivery/models/product_category.py
+++ b/addons/delivery/models/product_category.py
@@ -6,7 +6,7 @@ from odoo.exceptions import UserError
 
 
 class ProductCategory(models.Model):
-    _inherit = "product.category"
+    _inherit = ["product.category"]
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_delivery_category(self):

--- a/addons/delivery/models/res_partner.py
+++ b/addons/delivery/models/res_partner.py
@@ -4,6 +4,6 @@ from odoo import fields, models
 
 
 class ResPartner(models.Model):
-    _inherit = 'res.partner'
+    _inherit = ['res.partner']
 
     property_delivery_carrier_id = fields.Many2one('delivery.carrier', company_dependent=True, string="Delivery Method", help="Default delivery method used in sales orders.")

--- a/addons/delivery/models/sale_order.py
+++ b/addons/delivery/models/sale_order.py
@@ -7,7 +7,7 @@ from odoo.exceptions import UserError
 
 
 class SaleOrder(models.Model):
-    _inherit = 'sale.order'
+    _inherit = ['sale.order']
 
     pickup_location_data = fields.Json()
     carrier_id = fields.Many2one('delivery.carrier', string="Delivery Method", check_company=True, help="Fill this field if you plan to invoice the shipping based on picking.")

--- a/addons/delivery/models/sale_order_line.py
+++ b/addons/delivery/models/sale_order_line.py
@@ -4,7 +4,7 @@ from odoo import api, fields, models
 
 
 class SaleOrderLine(models.Model):
-    _inherit = 'sale.order.line'
+    _inherit = ['sale.order.line']
 
     is_delivery = fields.Boolean(string="Is a Delivery", default=False)
     product_qty = fields.Float(
